File: cs25backend/src/main/java/edu/dongguk/cs25backend/domain/type/OrderStatus.java
package edu.dongguk.cs25backend.domain.type;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum OrderStatus {
    //픽업대기, 픽업완료, 환불 처리
    READY("READY"), PICKUP("PICKUO"), CANCEL("CANCEL");

    private final String status;

    @Override
    public String toString() {
        return status;
    }
}

File: cs25server/src/main/kotlin/edu/dongguk/cs25server/domain/type/OrderStatus.kt
package edu.dongguk.cs25server.domain.type

import com.fasterxml.jackson.annotation.JsonCreator
import edu.dongguk.cs25server.exception.ErrorCode
import edu.dongguk.cs25server.exception.GlobalException

enum class OrderStatus(private val orderStatus: String) {
    READY("픽업대기"), PICKUP("픽업완료"), CANCEL("환불처리");

    override fun toString(): String {
        return orderStatus
    }

    companion object {
        @JvmStatic
        @JsonCreator
        fun getOrderStatus(orderStatus: String?): OrderStatus? {
            if (orderStatus.isNullOrEmpty())
                return null

            for(value in OrderStatus.values()) {
                if (orderStatus.equals(value.orderStatus)) {
                    return value
                }
            }
            throw GlobalException(ErrorCode.WRONG_ORDER_STATUS_ERROR)
        }
    }
}